### ISSUE TITLE
Issue with Docker MACVLAN and Traefik as ReverseProxy when using VisualEditor

### ISSUE THREAD

**Issue by Byolock** [CanastaWiki/Canasta#59]
Quick outline of my Setup :

- **Docker Host** is running on **192.168.1.10**
- A **non docker Webserver** is running on **192.168.1.10 port 5000** but also redirecting **port 80 and 443** to 5000
- **Traefik** is running on the docker host, **using MACVLAN on 192.168.1.2**
- **canasta** is running on the docker host with **port 80 mapped to 463** (so reachable by 192.168.1.10:463 without a reverse proxy)
- Traefik is configured to proxy `wiki.mydomain.com` to the canaste-web container on port 463
- The DNS Record `wiki.mydomain.com` points to 192.168.1.2 (the IP of the Traefik container)

This seems to be working just fine but if i try to edit anything in visual editor it does not load and only shows puzzle icons.
If I create a new wiki site and write something with visual editor and then hit save, it writes html source code into the article. Exactly that sourcecode from the non docker webserver. Somehow, instead of reading the source from the wiki page it seems to contact the webserver on 192.168.1.10 and simply uses the response of that. For Displaying that doesn't make any sense so it just doesn't show any meaningful data, saving on the other hand simply saves the raw html source code of that web server. 

If I set `$wgServer = "http://192.168.1.10:463";` and access it over this url it works fine. This strange behavior only starts when using the traefik proxy  and accessing over `wiki.mydomain.com`. 

So my uneducated guess would be it's looking for parsoid under the 192.168.1.10 IP and gets redirected to the webserver running on port 5000. I tried some host file entries redirecting `wiki.mydomain.com` to 'localhost', but that didn't work. I dont know there it should have that IP from and how to change this. Anybody got an idea?

To add even more strange things : 
- `ping wiki.mydomain.com`  from inside the container resolves to 172.20.0.3 => the docker network ip adress of that container
- `curl -v https://wiki.mydomain.com/rest.php/v1/page/Main_Page`  shows the response header `HTTP/1.1 302 Moved Temporarily < Server: nginx-dsm` => this is the http response header of the webserver running on 192.168.1.10:5000/443/80

**Comment by jeffw16** [comment]
I'm sorry to hear about these problems! While we are fine with keeping this issue open to see if anyone can get this fixed, Canasta does not offer support for setups that modify the deployment manifests in significant ways, such as modifying the ports and combining it with other Docker networks. Canasta is meant to run independently from existing services using the way we have set it up already.

**Comment by Byolock** [comment]
@jeffw16 
Thank you for your quick response. I completly understand. 

I just found out changing `$wgServer = "https://wiki.mydomain.com";` to `$wgServer = "http://192.168.1.10:463";` (changing https => http) does work. Traefik always forces https in the end anyway so I hope I don't create any security issue by using http in the config file. 
Pretty strange though that changing https to http leads me to a completly webserver running on a different IP address.

Anyway now that I actually got it in a working state and planning to rebuild my docker setup soon allowing me to build much simpler networking I will close this issue. Sorry to have taken up your time.
